services/hanke-service/src/main/resources/db/changelog/changesets/043-change-hanke-perustaja-non-nullable.yaml

```diff
@@ -0,0 +1,21 @@
+databaseChangeLog:
+  - changeSet:
+      id: 043-add-not-null-constraint-for-hanke-perustaja
+      author: Niko Pitkonen
+      changes:
+        - addDefaultValue:
+            tableName: hanke
+            columnName: perustajanimi
+            defaultValue: ""
+        - addDefaultValue:
+            tableName: hanke
+            columnName: perustajaemail
+            defaultValue: ""
+        - addNotNullConstraint:
+            tableName: hanke
+            columnName: perustajanimi
+            defaultNullValue: ""
```

Comment:
I don't understand why we're changing a clear indicator for a missing value (null value) to an unclear one.

Also for the name, we know we can't get the name in every situation.

Reply:
Would it be better to remove the constraint on the db / entity level? I.e. keep the objects as they are and just validate the the request in the controller?

Reply:
If we want to add the constraints at least on the email, we could wipe the current hanke_kayttaja and kayttaja_tunniste. They're not really usable anyway.

Having a tight constraint on the name will cause problems when we use Helsinki AD authentication (like we do currently in the dev environment).

Reply:
Yes, I think we can't add the constraints to DB. We can't figure out the founder after the fact, so the unknown ones should be null. Or maybe we could do a complex query to find the contact with the orderer flag from every hanke since all hanke in production are cable report applications. 

Having a tight constraint on the name might cause problems when we use Helsinki AD authentication (like we do currently in the dev environment, and probably for local development).